app/src/test/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationPresenterImplSpec.kt

```diff
@@ -0,0 +1,51 @@
+package io.homeassistant.companion.android.onboarding.integration
+
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import io.mockk.coVerifyAll
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+
+object MobileAppIntegrationPresenterImplSpec : Spek({
+
+    beforeEachTest {
+        Dispatchers.setMain(Dispatchers.Unconfined)
+    }
+
+    afterEachTest {
+        Dispatchers.resetMain()
+    }
+
+    describe("presenter") {
+        val integrationUseCase by memoized { mockk<IntegrationUseCase>(relaxUnitFun = true) }
+        val view by memoized { mockk<MobileAppIntegrationView>(relaxUnitFun = true) }
+        val presenter by memoized { MobileAppIntegrationPresenterImpl(view, integrationUseCase) }
+
+        describe("on click skip") {
+            beforeEachTest {
+                presenter.onSkip()
+            }
+
+            it("should continue") {
+                coVerifyAll { view.registrationSkipped() }
+            }
+        }
+
+        describe("on click retry") {
```

Comment:
I know you don't like write test but we need it. 
Could you add all the test case : 
integrationUseCase.registerDevice 'on success' => onRegistrationAttempt = Display loading && view deviceRegistered
integrationUseCase.registerDevice 'on error' => onRegistrationAttempt = Display loading && display error

Reply:
Already on it.  I know the value of them, just reminds me a lot of my day job, haha